Description: Enable non-default python versions for test runs
 The original version of runtests.sh only allows to run the tests on the current 
 default python version. This patch add the environment variable PYTHON that can 
 be set to a specific python binary (e.g. python2.6) to run the tests with.
 It also changes the path for django-admin.py to the one supplied by the 
 python-django package.
Author: Michael Fladischer <FladischerMichael@fladi.at>
Last-Update: 2011-11-26
Forwarded: no

--- a/runtests.sh
+++ b/runtests.sh
@@ -1,6 +1,9 @@
 #!/bin/sh
 
 export PYTHONPATH=.
+if [ -z $PYTHON ]; then
+    export PYTHON=python
+fi
 
 django_test() {
     TEST="$1"
@@ -25,9 +28,8 @@
     fi
 }
 
-django_test 'django-admin.py test --settings=testapp.settings' '2' 'normal settings'
-django_test 'django-admin.py test --settings=testapp.settings_with_south' '2' 'with south in installed apps'
-django_test 'django-admin.py test --settings=testapp.settings_old_style' '2' 'django_nose.run_tests format'
-django_test 'testapp/runtests.py testapp.test_only_this' '1' 'via run_tests API'
-django_test 'django-admin.py test --settings=testapp.settings_with_plugins testapp/plugin_t' '1' 'with plugins'
-django_test 'django-admin.py test --settings=testapp.settings unittests' '4' 'unittests'
+django_test "$PYTHON /usr/bin/django-admin test --settings=testapp.settings" '2' 'normal settings'
+django_test "$PYTHON /usr/bin/django-admin test --settings=testapp.settings_with_south" '2' 'with south in installed apps'
+django_test "$PYTHON /usr/bin/django-admin test --settings=testapp.settings_old_style" '2' 'django_nose.run_tests format'
+django_test "$PYTHON testapp/runtests.py testapp.test_only_this" '1' 'via run_tests API'
+django_test "$PYTHON /usr/bin/django-admin test --settings=testapp.settings_with_plugins testapp/plugin_t" '1' 'with plugins'
